edit/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+use std::ffi::OsString;
+use std::path::Path;
+use std::str::FromStr;
+
+use radicle_common::args::{Args, Error, Help};
+use radicle_common::{git, keys, profile};
+use radicle_terminal as term;
+
+use librad::git::identities::{any, person, project};
+use librad::git::Urn;
+
+use link_identities::payload::{PersonPayload, ProjectPayload};
+
+use anyhow::anyhow;
+
+pub const HELP: Help = Help {
+    name: "edit",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad edit [<urn>] [<option>...]
+
+    Edits the identity document pointed to by the URN. If it isn't specified,
+    the current project is edited.
+
+Options
+
+    --help              Print help
+"#,
+};
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Options {
+    pub urn: Option<Urn>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut urn: Option<Urn> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if urn.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Urn::from_str(&val) {
+                        urn = Some(val);
+                    } else {
+                        return Err(anyhow!("invalid URN '{}'", val));
+                    }
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options { urn }, vec![]))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+
+    let urn = if let Some(urn) = options.urn {
+        urn
+    } else {
+        let repo = git::Repository::open(Path::new("."))?;
+
+        radicle_common::project::rad_remote(&repo)?.url.urn
+    };
+
+    let identity = any::get(&storage, &urn)?
+        .ok_or_else(|| anyhow!("No project or person found for this URN"))?;
+
+    if let Some(updated_payload) = term::Editor::new()
+        .edit(&serde_json::to_string_pretty(&identity.payload())?)
+        .unwrap()
+    {
+        if identity.project().is_some() {
+            let payload: ProjectPayload = serde_json::from_str(&updated_payload)?;
+            project::update(&storage, &urn, None, payload, None)?;
+        } else {
+            let payload: PersonPayload = serde_json::from_str(&updated_payload)?;
+            person::update(&storage, &urn, None, payload, None)?;
+        }
+        term::info!("Updated!");
```

Comment:
`term::success("Update successful")`

Reply:
https://github.com/radicle-dev/radicle-cli/commit/ec0b9679b7a5e43fc48691fca49f262e317765bb#diff-7f6d0944507bd7f26b50af3f4bce4555849243552f506948e0acbbffe597f4f8R102

Reply:
also, `success!(...)` is a macro.

Is there a logic behind which ones are fns e.g. `term::warning(...)` and the ones which are macro?

We should create macros for fns, just to make it simpler for devs to use.